File: app/lib/core/feedback.dart
/// Shared feedback and report labels (parity with src/utils/feedback.js).
library;

class FeedbackCategory {
  const FeedbackCategory(this.id, this.label);
  final String id;
  final String label;
}

class ReportReason {
  const ReportReason(this.id, this.label);
  final String id;
  final String label;
}

const feedbackCategories = [
  FeedbackCategory('feature', 'Feature idea'),
  FeedbackCategory('improvement', 'Improvement'),
  FeedbackCategory('bug', 'Bug'),
  FeedbackCategory('other', 'Other'),
];

const reportReasons = [
  ReportReason('spam', 'Spam'),
  ReportReason('wrong_info', 'Wrong info'),
  ReportReason('offensive', 'Offensive'),
  ReportReason('duplicate', 'Duplicate'),
  ReportReason('other', 'Other'),
];

const feedbackMinMessageLen = 10;
const feedbackMaxMessageLen = 4000;
const reportMaxDetailsLen = 2000;

bool isFeedbackSubmittable(String message) {
  final trimmed = message.trim();
  return trimmed.length >= feedbackMinMessageLen &&
      trimmed.length <= feedbackMaxMessageLen;
}

String feedbackErrorMessage(String? code) {
  switch (code) {
    case 'message_too_short':
      return 'Please write at least 10 characters.';
    case 'message_too_long':
      return 'Message is too long.';
    case 'rate_limited':
      return 'Too many submissions recently. Try again later.';
    case 'database_unavailable':
      return 'Service temporarily unavailable.';
    default:
      return 'Could not send feedback. Try again.';
  }
}

String reportErrorMessage(String? code) {
  switch (code) {
    case 'duplicate_report':
      return 'You already reported this. We will follow up if needed.';
    case 'not_authenticated':
      return 'Sign in to report content.';
    case 'target_not_found':
      return 'This content could not be found. It may have been removed.';
    case 'invalid_reason':
      return 'Choose a reason for your report.';
    case 'database_unavailable':
      return 'Service temporarily unavailable.';
    default:
      return 'Could not send report. Try again.';
  }
}

String reportContextForEvent({
  required String courseCode,
  required String title,
  required String date,
}) => '$courseCode · $title · $date';

String reportContextForPolicy(String courseCode) => '$courseCode · course policy';

String reportContextForOccupiedRoom({
  required String room,
  required String date,
}) => '$room · $date';

File: src/utils/feedback.js
export const FEEDBACK_CATEGORIES = [
    { id: 'feature', label: 'Feature idea' },
    { id: 'improvement', label: 'Improvement' },
    { id: 'bug', label: 'Bug' },
    { id: 'other', label: 'Other' },
];

export const REPORT_REASONS = [
    { id: 'spam', label: 'Spam' },
    { id: 'wrong_info', label: 'Wrong info' },
    { id: 'offensive', label: 'Offensive' },
    { id: 'duplicate', label: 'Duplicate' },
    { id: 'other', label: 'Other' },
];

export const FEEDBACK_MIN_MESSAGE_LEN = 10;
export const FEEDBACK_MAX_MESSAGE_LEN = 4000;
export const REPORT_MAX_DETAILS_LEN = 2000;

export function isFeedbackSubmittable(message) {
    const trimmed = String(message || '').trim();
    return (
        trimmed.length >= FEEDBACK_MIN_MESSAGE_LEN
        && trimmed.length <= FEEDBACK_MAX_MESSAGE_LEN
    );
}

export function feedbackErrorMessage(code) {
    switch (code) {
        case 'message_too_short':
            return 'Please write at least 10 characters.';
        case 'message_too_long':
            return 'Message is too long.';
        case 'rate_limited':
            return 'Too many submissions recently. Try again later.';
        case 'database_unavailable':
            return 'Service temporarily unavailable.';
        default:
            return 'Could not send feedback. Try again.';
    }
}

export function reportErrorMessage(code) {
    switch (code) {
        case 'duplicate_report':
            return 'You already reported this. We will follow up if needed.';
        case 'not_authenticated':
            return 'Sign in to report content.';
        case 'target_not_found':
            return 'This content could not be found. It may have been removed.';
        case 'invalid_reason':
            return 'Choose a reason for your report.';
        case 'database_unavailable':
            return 'Service temporarily unavailable.';
        default:
            return 'Could not send report. Try again.';
    }
}

export function reportContextLabel(snapshot, fallback = '') {
    if (!snapshot) return fallback;
    if (snapshot.kind === 'course_event') {
        return `${snapshot.courseCode} · ${snapshot.title} · ${snapshot.date}`;
    }
    if (snapshot.kind === 'course_policy') {
        return `${snapshot.courseCode} · course policy`;
    }
    if (snapshot.kind === 'occupied_room') {
        return `${snapshot.room} · ${snapshot.date}`;
    }
    if (snapshot.kind === 'other') {
        return snapshot.label || snapshot.pageContext || fallback;
    }
    return fallback;
}
